src/cmd-buildextend-installer

```diff
@@ -177,54 +191,28 @@ def generate_iso():
         stamppath = os.path.join(tmproot, stampname)
         os.makedirs(os.path.dirname(stamppath), exist_ok=True)
         open(stamppath, 'w').close()
-        mkinitrd(tmproot, initramfs_cpio_ext)
-
-    # Append the "stampfile initramfs" to the base, also
-    # copying the base initramfs.
-    tmp_initramfs = os.path.join(tmpdir, 'initramfs')
-    with open(tmp_initramfs, 'wb') as fdst:
-        with open(initramfs, 'rb') as fsrc:
-            shutil.copyfileobj(fsrc, fdst)
-        with open(initramfs_cpio_ext, 'rb') as fsrc:
-            shutil.copyfileobj(fsrc, fdst)
-    os.rename(tmp_initramfs, initramfs)
-    os.unlink(initramfs_cpio_ext)
+        extend_initrd(base_initramfs, tmproot)
 
+    # The base_initramfs from here is shared between the ISO and PXE paths
+    cp_reflink(base_initramfs, initramfs)
+    tmp_squashfs = None
     if is_live:
-        tmp_initramfs = os.path.join(tmpdir, 'initramfs')
-        osmet_initramfs = None
         if osmet:
-            osmet_initramfs = os.path.join(tmpdir, 'initramfs-osmet.img')
             with tempfile.TemporaryDirectory(prefix='initramfs', dir=tmpdir) as tmproot:
                 tmp_osmet = os.path.join(tmproot, img_metal_obj['path'] + '.osmet')
                 print(f'Generating osmet file')
                 run_verbose(['/usr/lib/coreos-assembler/osmet-pack',
                             img_metal, tmp_osmet, img_metal_checksum])
-                mkinitrd(tmproot, osmet_initramfs, compression=False)
-
-        # The initramfs image is a hardlink to the uncompressed objects
-        # cache, so we can't modify it in place.
-        with open(tmp_initramfs, 'wb') as fdst:
-            with open(initramfs, 'rb') as fsrc:
-                shutil.copyfileobj(fsrc, fdst)
-            # Append osmet if we have it
-            if osmet_initramfs is not None:
-                with open(osmet_initramfs, 'rb') as fsrc:
-                    shutil.copyfileobj(fsrc, fdst)
-            fdst.flush()
-            # Create the squashfs for the root, and then as soon as possible
-            # insert it inline into the initramfs.  This avoids
-            # copying the squashfs multiple times.  Ideally in the future
-            # we directly write it into the cpio archive.
-            with tempfile.TemporaryDirectory(prefix='initramfs', dir=tmpdir) as tmproot:
-                print(f'Compressing squashfs with {squashfs_compression}')
-                run_verbose(['/usr/lib/coreos-assembler/gf-mksquashfs',
-                            img_metal, os.path.join(tmproot, 'root.squashfs'), squashfs_compression])
-                # Compression is redundant but the kernel requires it
-                mkinitrd_pipe(tmproot, fdst, compression=False)
-            # And finally, the padding
+                extend_initrd(base_initramfs, tmproot, compression=False)
```

Comment:
Hmm, OK so I *think* there's an issue here. I think we need to append osmet to the base initramfs *before* `cp_reflink(base_initramfs, initramfs)`, right? Otherwise, PXE will include osmet, but not the live ISO initrd.

Reply:
Ahh I think that might be what CI is breaking on:

```
+ env TMPDIR=/srv/tmp/ kola testiso -S --scenarios iso-offline-install
Testing scenarios: [iso-offline-install]
Error: scenario iso-offline-install: timed out after 15m0s
```

Reply:
I guess there's an interesting topic here of whether it's worth moving osmet out of the initramfs (and into the ISO root) in the ISO path too.  Probably not, it's small.

But it's conceptually not needed for booting either.

Reply:
> I guess there's an interesting topic here of whether it's worth moving osmet out of the initramfs (and into the ISO root) in the ISO path too. Probably not, it's small.
>
> But it's conceptually not needed for booting either.

Yeah, true. I added some comments related to that in https://github.com/coreos/fedora-coreos-tracker/issues/390#issuecomment-626773741 and following. I'd say let's just leave it as appended to the base initrd for now? (One nice thing about that is that it's then handled by the existing code that handles both ISO and PXE, which I think is a win overall given that it's such a small cost anyway).

Reply:
Yep I'm fine with that.